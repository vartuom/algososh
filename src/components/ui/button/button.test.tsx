import renderer from 'react-test-renderer';
import {Button} from "./button";
import {fireEvent, render, screen} from "@testing-library/react";

describe('Button test', () => {
    it('renders with text', () => {
        const buttonElement = renderer.create(<Button text={'some text'} />).toJSON();
        expect(buttonElement).toMatchSnapshot();
    })
    it('renders without text', () => {
        const buttonElement = renderer.create(<Button />).toJSON();
        expect(buttonElement).toMatchSnapshot();
    })
    it('renders with disabled prop', () => {
        const buttonElement = renderer.create(<Button disabled={true}/>).toJSON();
        expect(buttonElement).toMatchSnapshot();
    })
    it('renders with loader prop', () => {
        const buttonElement = renderer.create(<Button isLoader={true}/>).toJSON();
        expect(buttonElement).toMatchSnapshot();
    })
    it('fires callback on click', () => {
        const callbackFn  = jest.fn();
        render(<Button text={'callbackTest'} onClick={callbackFn}/>);
        const buttonElement = screen.getByText('callbackTest');
        fireEvent.click(buttonElement);
        expect(callbackFn).toHaveBeenCalledTimes(1);
    })
})