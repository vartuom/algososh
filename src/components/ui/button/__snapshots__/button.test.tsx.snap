// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test renders with disabled prop 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button test renders with loader prop 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test renders with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    some text
  </p>
</button>
`;

exports[`Button test renders without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
